### ISSUE TITLE
failing to build 0.9.1

### ISSUE THREAD

**Issue by derdigge** [neutrinolabs/xrdp#602]
Hello! 

I am not able to build xrdp in ubuntu xenial. configure fails with:

```
checking for IceConnectionNumber in -lICE... yes
checking for X11/extensions/Xfixes.h... yes
checking for X11/extensions/Xrandr.h... no
configure: error: please install libxrandr-dev or libXrandr-devel
```

so i did:

```
root@ubuntu-desktop:~/xrdp# apt-file search X11/extensions/Xrandr.h
libxrandr-dev: /usr/include/X11/extensions/Xrandr.h
root@ubuntu-desktop:~/xrdp# apt install libxrandr-dev
Paketlisten werden gelesen... Fertig
Abhängigkeitsbaum wird aufgebaut.
Statusinformationen werden eingelesen.... Fertig
»libxrandr-dev« ist bereits die neuste Version (2:1.5.0-1).
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
```

so it tells me it is allready latest.

So i installed the latest libxrandr from http://packages.ubuntu.com/zesty/libxrandr-dev
with the same result.

the build o matic is doing the same errors and xenial buildin 0.6.1-2 is also broken. 

Whats on here ?

EDIT:

I installed a ready made deb from here:
http://packages.ubuntu.com/zesty/xrdp
http://packages.ubuntu.com/zesty/xorgxrdp

installs and starts okay. Maybe this helps someone in the furute.

Anyhow xrdp isnt working with xorg or x11vnc.

xrdp.log:
```
[20161230-19:58:02] [INFO ] A connection received from: ::ffff:10.3.0.6 port 59924
[20161230-19:58:02] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.6.20 port 3389)
[20161230-19:58:02] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20161230-19:58:02] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20161230-19:58:02] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20161230-19:58:02] [DEBUG] Security layer: requested 11, selected 1
[20161230-19:58:02] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.6.20 port 3389)
[20161230-19:58:02] [ERROR] Listening socket is in wrong state, terminating listener
[20161230-19:58:02] [INFO ] A connection received from: ::ffff:10.3.0.6 port 59925
[20161230-19:58:02] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20161230-19:58:02] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.6.20 port 3389)
[20161230-19:58:02] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20161230-19:58:02] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20161230-19:58:02] [DEBUG] Security layer: requested 11, selected 1
[20161230-19:58:03] [DEBUG] xrdp_00004daf_wm_login_mode_event_00000001
[20161230-19:58:03] [INFO ] Loading keymap file /etc/xrdp/km-00000407.ini
[20161230-19:58:03] [WARN ] local keymap file for 0x00000407 found and doesn't match built in keymap, using local keymap file
[20161230-19:58:23] [DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350
[20161230-19:58:23] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20161230-19:58:23] [DEBUG] xrdp_wm_log_msg: sending login info to session manager, please wait...
[20161230-19:58:23] [DEBUG] return value from xrdp_mm_connect 0
[20161230-19:58:23] [INFO ] xrdp_wm_log_msg: login successful for display 10
[20161230-19:58:23] [DEBUG] xrdp_wm_log_msg: started connecting
[20161230-19:58:27] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:30] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:34] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:38] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:41] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:45] [DEBUG] Closed socket 19 (AF_UNIX)
[20161230-19:58:48] [DEBUG] Closed socket 19 (AF_UNIX)
```

sesman.log:
```
[20161230-20:00:03] [INFO ] A connection received from ::1 port 56168
[20161230-20:00:04] [INFO ] ++ created session (access granted): username derdigge, ip 0.0.0.0:59941 - socket: 12
[20161230-20:00:04] [INFO ] starting Xorg session...
[20161230-20:00:04] [DEBUG] Closed socket 9 (AF_INET6 :: port 5910)
[20161230-20:00:04] [DEBUG] Closed socket 9 (AF_INET6 :: port 6010)
[20161230-20:00:04] [DEBUG] Closed socket 9 (AF_INET6 :: port 6210)
[20161230-20:00:04] [DEBUG] Closed socket 7 (AF_INET6 ::1 port 3350)
[20161230-20:00:04] [DEBUG] Closed socket 8 (AF_INET6 ::1 port 3350)
[20161230-20:00:04] [INFO ] Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp
[20161230-20:00:14] [ERROR] X server for display 10 startup timeout
[20161230-20:00:14] [INFO ] starting xrdp-sessvc - xpid=19921 - wmpid=19920
[20161230-20:00:14] [ERROR] X server for display 10 startup timeout
[20161230-20:00:14] [ERROR] another Xserver might already be active on display 10 - see log
[20161230-20:00:14] [DEBUG] aborting connection...
[20161230-20:00:14] [INFO ] ++ terminated session:  username derdigge, display :10.0, session_pid 19919, ip 0.0.0.0:59941 - socket: 12
```

next i tried to launch a log on my own by modifiing pram(s) in sesman.ini an tried like 

```
[20161230-20:32:29] [INFO ] A connection received from ::1 port 56216
[20161230-20:32:30] [INFO ] ++ created session (access granted): username derdigge, ip 0.0.0.0:60391 - socket: 12
[20161230-20:32:30] [INFO ] starting Xorg session...
[20161230-20:32:30] [DEBUG] Closed socket 9 (AF_INET6 :: port 5910)
[20161230-20:32:30] [DEBUG] Closed socket 9 (AF_INET6 :: port 6010)
[20161230-20:32:30] [DEBUG] Closed socket 9 (AF_INET6 :: port 6210)
[20161230-20:32:30] [DEBUG] Closed socket 7 (AF_INET6 ::1 port 3350)
[20161230-20:32:30] [DEBUG] Closed socket 8 (AF_INET6 ::1 port 3350)
[20161230-20:32:30] [INFO ] Xorg :10 -auth .Xauthority -config /etc/X11/xrdp/xorg.conf -noreset -nolisten tcp -logfile /var/log/xrdp_X10.log
[20161230-20:32:40] [ERROR] X server for display 10 startup timeout
[20161230-20:32:40] [INFO ] starting xrdp-sessvc - xpid=22413 - wmpid=22412
[20161230-20:32:40] [ERROR] X server for display 10 startup timeout
[20161230-20:32:40] [ERROR] another Xserver might already be active on display 10 - see log
[20161230-20:32:40] [DEBUG] aborting connection...
[20161230-20:32:40] [INFO ] ++ terminated session:  username derdigge, display :10.0, session_pid 22411, ip 0.0.0.0:60391 - socket: 12
```

/var/log/xrdp_X10.log gets not created. Something is wrong with the "sesmen starts xorg" mechanics. I do not know enough about how sesman is doing stuff so please help.

any other thoughts?

**Comment by proski** [comment]
I actually compiled the current devel branch on Ubuntu 16.04 some time ago (around the 0.9.1 release), and it worked correctly for me. Could you check `config.log` for details of the failure to find X11/extensions/Xrandr.h?

You are using a non-standard location for the Xorg log. It may not have permission to write the log. The standard configuration files would tell Xorg to use your home directory.

I believe Xvnc has issues with IPv6 support. I haven't had a chance to look deeper into that.

**Comment by moobyfr** [comment]
you can follow the wiki page https://github.com/neutrinolabs/xrdp/wiki/Building-on-Debian-8 on ubuntu, I took it exactly to build on ubuntu today for a 16.10. I can have a look again for xenial if needed.
0.6 as you wrote, isn't supported anymore better have a stable release as 0.9.1 I don't know the version for RDP-O-Matic.

**Comment by derdigge** [comment]
@moobyfr 
Thanks. the default link in the wiki points to Debian 6. Build didnt work for me with this one, dont know why anymore.  Maybe you want to change that link for others to come.

**Comment by proski** [comment]
I tried and could not reproduce the issue. I launched an Ubuntu 16.04.1 instance on AWS and compiled xrdp-0.9.1 from the actual release tarball without any issues.

**Comment by derdigge** [comment]
okay. i did a local KVM and compiling wents fine now on Ubuntu-Gnome. I fixed wrong path and $vars in the wiki here https://github.com/neutrinolabs/xrdp/wiki/Building-on-Debian-8


Anyhow. On a Clean install it is leading into the thame errors. Something is misfiged by default? 
Here my logs.

session type x11rdp:

```
[20161231-10:58:04] [INFO ] A connection received from 127.0.0.1 port 44932
[20161231-10:58:05] [INFO ] ++ created session (access granted): username derdigge, ip 10.3.0.6:65082 - socket: 12
[20161231-10:58:05] [INFO ] starting X11rdp session...
[20161231-10:58:05] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:5910)
[20161231-10:58:05] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6010)
[20161231-10:58:05] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6210)
[20161231-10:58:05] [DEBUG] Closed socket 8 (AF_INET 127.0.0.1:3350)
[20161231-10:58:05] [DEBUG] Closed socket 7 (AF_INET 127.0.0.1:3350)
[20161231-10:58:05] [INFO ] X11rdp :10 -auth .Xauthority -geometry 1920x1080 -depth 24 -bs -nolisten tcp -uds
[20161231-10:58:05] [CORE ] error starting X server - user derdigge - pid 16390
[20161231-10:58:05] [DEBUG] errno: 2, description: No such file or directory
[20161231-10:58:05] [DEBUG] execve parameter list size: 13
[20161231-10:58:05] [DEBUG]         argv[0] = X11rdp
[20161231-10:58:05] [DEBUG]         argv[1] = :10
[20161231-10:58:05] [DEBUG]         argv[2] = -auth
[20161231-10:58:05] [DEBUG]         argv[3] = .Xauthority
[20161231-10:58:05] [DEBUG]         argv[4] = -geometry
[20161231-10:58:05] [DEBUG]         argv[5] = 1920x1080
[20161231-10:58:05] [DEBUG]         argv[6] = -depth
[20161231-10:58:05] [DEBUG]         argv[7] = 24
[20161231-10:58:05] [DEBUG]         argv[8] = -bs
[20161231-10:58:05] [DEBUG]         argv[9] = -nolisten
[20161231-10:58:05] [DEBUG]         argv[10] = tcp
[20161231-10:58:05] [DEBUG]         argv[11] = -uds
[20161231-10:58:05] [DEBUG]         argv[12] = (null)
[20161231-10:58:15] [ERROR] X server for display 10 startup timeout
[20161231-10:58:15] [INFO ] starting xrdp-sessvc - xpid=16390 - wmpid=16389
[20161231-10:58:15] [ERROR] X server for display 10 startup timeout
[20161231-10:58:15] [ERROR] another Xserver might already be active on display 10 - see log
[20161231-10:58:15] [DEBUG] aborting connection...
[20161231-10:58:15] [INFO ] ++ terminated session:  username derdigge, display :10.0, session_pid 16388, ip 10.3.0.6:65082 - socket: 12
```
No such file or directory? Yeah but what is he looking for? If we find out i would add that one to the wiki page.

session type Xorg:

```
[20161231-11:00:19] [INFO ] A connection received from 127.0.0.1 port 44934
[20161231-11:00:19] [INFO ] ++ created session (access granted): username derdigge, ip 10.3.0.6:65124 - socket: 12
[20161231-11:00:19] [INFO ] starting Xorg session...
[20161231-11:00:19] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:5910)
[20161231-11:00:19] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6010)
[20161231-11:00:19] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6210)
[20161231-11:00:19] [DEBUG] Closed socket 7 (AF_INET 127.0.0.1:3350)
[20161231-11:00:19] [DEBUG] Closed socket 8 (AF_INET 127.0.0.1:3350)
[20161231-11:00:19] [INFO ] Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log
[20161231-11:00:29] [ERROR] X server for display 10 startup timeout
[20161231-11:00:29] [INFO ] starting xrdp-sessvc - xpid=16403 - wmpid=16402
[20161231-11:00:29] [ERROR] X server for display 10 startup timeout
[20161231-11:00:29] [ERROR] another Xserver might already be active on display 10 - see log
[20161231-11:00:29] [DEBUG] aborting connection...
[20161231-11:00:29] [INFO ] ++ terminated session:  username derdigge, display :10.0, session_pid 16401, ip 10.3.0.6:65124 - socket: 12
```

Same error as above. Where is this logfile stored? i cant find it.

```
root@Desktop:/var/log# find / | grep 'xorgxrdp'
find: ‘/run/user/1000/gvfs’: Permission denied
/usr/lib/xorg/modules/libxorgxrdp.la
/usr/lib/xorg/modules/libxorgxrdp.a
/usr/lib/xorg/modules/libxorgxrdp.so
```

is still cannot start Xorg with Xorg session and there is a mystic "file not found" The loglevel is on debug i prupose to make that level more verbose to tell me what it is trying to find.

**Comment by moobyfr** [comment]
There are several backend for Xrdp: X11rdp, Xorg and Xvnc. X11rdp isn't installed by default, you'll have to build it . Xorg is the preferred backend to use.

If you are using ubuntu, you have probably hit another bug: bug #598

**Comment by derdigge** [comment]
@moobyfr 
lets see i build x11rdp now. But why is Xorg session not working? Xorg is definitly there :)

**Comment by derdigge** [comment]
Building of X11R7.8 fails. dieing:

```
make[4]: Entering directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3/hw/xfree86/loader'
CPP='gcc -E' AWK='mawk' ./sdksyms.sh ../../.. -DHAVE_DIX_CONFIG_H -Wall -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wnested-externs -fno-strict-aliasing -Wbad-function-cast -Wformat=2  -D_BSD_SOURCE -DHAS_FCHOWN -DHAS_STICKY_DIR_BIT -I/usr/local/include -I/usr/local/include/pixman-1 -I/usr/local/include -I/usr/local/include/freetype2 -I/usr/local/include -I../../../include -I../../../include -I../../../Xext -I../../../composite -I../../../damageext -I../../../xfixes -I../../../Xi -I../../../mi -I../../../miext/shadow  -I../../../miext/damage -I../../../render -I../../../randr -I../../../fb -fvisibility=hidden -DHAVE_XORG_CONFIG_H -fvisibility=hidden -I/usr/local/include -DXF86PM -I/usr/local/include -I/usr/local/include -fPIC -O2 -I../../../hw/xfree86 -I../../../hw/xfree86/include -I../../../hw/xfree86/common -I../../../hw/xfree86/os-support -I../../../hw/xfree86/os-support/bus -I../../../os -I./../parser -I../../../miext/cw -I./../ddc -I./../i2c -I./../modes -I./../ramdac
  CC     loader.lo
  CC     loadext.lo
  CC     loadmod.lo
  CC     dlloader.lo
loadmod.c: In function ‘FreeSubdirs’:
loadmod.c:378:11: warning: passing argument 1 of ‘free’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
      free(*s);
           ^
In file included from ../../../include/misc.h:142:0,
                 from ../../../include/os.h:52,
                 from loadmod.c:53:
/usr/include/stdlib.h:483:13: note: expected ‘void *’ but argument is of type ‘const char *’
 extern void free (void *__ptr) __THROW;
             ^
  CC     os.lo
  CC     sdksyms.lo
sdksyms.c:323:15: error: expected expression before ‘,’ token
     (void *) &,
               ^
Makefile:512: recipe for target 'sdksyms.lo' failed
make[4]: *** [sdksyms.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3/hw/xfree86/loader'
Makefile:700: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3/hw/xfree86'
Makefile:562: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3/hw/xfree86'
Makefile:486: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3/hw'
Makefile:624: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
make: Leaving directory '/root/git/neutrinolabs/xrdp-0.9.1/xorg/X11R7.6/build_dir/xorg-server-1.9.3'
```
it is related  to xfree loader. I saw your posting in the wiki. 

"If you don't want to use the xorgxrdp module, you can still build a local Xserver dedicated to xrdp:"

how do i use xorgxrdp as it is allready installed during compile process.

**Comment by derdigge** [comment]
I am asking because your wiki is telling me here how it works:
https://github.com/neutrinolabs/xrdp/wiki/Compiling-and-using-xorgxrdp

which also fails to start X  :)

`Dec 31 12:11:45 Desktop xrdp-sesman[3942]: (3942)(140502160410368)[CORE ] error starting X server - user derdigge - pid 3942`

Sorry for bothering arround here but i am sure that many users simply give up at that point. It is a clean installation with straight forward implementation. Maybe it is working "on your machine" but a clean install did not work for most.

**Comment by moobyfr** [comment]
As I commented before: 
If you are using ubuntu, you have probably hit another bug: bug #598

I use it on production on Debian for users, there is no problem. Ubuntu is another problem.

**Comment by moobyfr** [comment]
I looked at your error: 
as the log is seing, your wrapper isn't executable.

**Comment by proski** [comment]
It is known that X11rdp is hard to compile. It is a collection of 101 packages with different requirements and quirks. There is no master configure script for it that would check all dependencies in advance. Perhaps it should be added.

Apparently, sdksyms.c generation is failing for you. I'm trying to figure out why.

**Comment by derdigge** [comment]
@moobyfr 
Thanks for your time an patience :)

missing to make it accesible was an accident but did not solve my issue. I enabled rootlogins for that kvm box in case there are "permission issues" without success. I still see a startup timeout. This is with /usr/bin/XorgXrdp from usage section of https://github.com/neutrinolabs/xrdp/wiki/Compiling-and-using-xorgxrdp

```
[20170101-14:27:32] [INFO ] A connection received from 127.0.0.1 port 60696
[20170101-14:27:32] [INFO ] ++ created session (access granted): username root, ip 10.3.0.6:63506 - socket: 12
[20170101-14:27:32] [INFO ] starting Xorg session...
[20170101-14:27:32] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:5910)
[20170101-14:27:32] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6010)
[20170101-14:27:32] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6210)
[20170101-14:27:32] [DEBUG] Closed socket 8 (AF_INET 127.0.0.1:3350)
[20170101-14:27:32] [DEBUG] Closed socket 7 (AF_INET 127.0.0.1:3350)
[20170101-14:27:32] [INFO ] XorgXrdp :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log
[20170101-14:27:42] [ERROR] X server for display 10 startup timeout
[20170101-14:27:42] [INFO ] starting xrdp-sessvc - xpid=1361 - wmpid=1360
[20170101-14:27:42] [ERROR] X server for display 10 startup timeout
[20170101-14:27:42] [ERROR] another Xserver might already be active on display 10 - see log
[20170101-14:27:42] [DEBUG] aborting connection...
[20170101-14:27:42] [INFO ] ++ terminated session:  username root, display :10.0, session_pid 1359, ip 10.3.0.6:63506 - socket: 12
```

I do not have the enough Understanding of xorg to reproduce "setting suid to Xorg" like described in #598 

I will install an Debian now in local KVM for testing.

**Comment by metalefty** [comment]
The original issue seems to be solved, closing.
